engine/src/main/java/org/hibernate/validator/messageinterpolation/ResourceBundleMessageInterpolator.java

```diff
@@ -200,11 +199,16 @@ private static ExpressionFactory buildExpressionFactory() {
 
 			// Finally we try the CL of the EL implementation itself. This is necessary for OSGi now that the
 			// implementation is separated from the API.
-			run( SetContextClassLoader.action( ExpressionFactoryImpl.class.getClassLoader() ) );
-			if ( canLoadExpressionFactory() ) {
-				ExpressionFactory expressionFactory = ELManager.getExpressionFactory();
-				LOG.debug( "Loaded expression factory via com.sun.el classloader" );
-				return expressionFactory;
+			try {
+				run( SetContextClassLoader.action( Class.forName( "com.sun.el.ExpressionFactoryImpl" ).getClassLoader() ) );
```

Comment:
Hey, thanks for submitting this PR.

```java
    @CallerSensitive
    public static Class<?> forName(String className)
                throws ClassNotFoundException {
        Class<?> caller = Reflection.getCallerClass();
        return forName0(className, true, ClassLoader.getClassLoader(caller), caller);
    }
```
Since that's how the `Class.forName` is implemented wouldn't we just end up having the same class loader as in the very first try of this `buildExpressionFactory()`? This part:
```java
if ( canLoadExpressionFactory() ) {
			ExpressionFactory expressionFactory = ELManager.getExpressionFactory();
			LOG.debug( "Loaded expression factory via original TCCL" );
			return expressionFactory;
		}
```

It would also be nice if the changes were accompanied by a test case; thanks!

Reply:
I would refrain from changing the logic in there. It is very brittle and we worked around multiple issues with OSGi and the modular classpath from WildFly.

Reply:
I've revert to `ExpressionFactoryImpl.class.getClassLoader()`, just catch `NoClassDefFoundError` and ignore it.